### ISSUE TITLE
__name__ when executing a script

### ISSUE THREAD

**Issue by jccraig** [numworks/epsilon#1281]
In standard Python, when a script is imported, the __name__ variable is set to the name of the script's file name.  But when a script is executed as the main script, its __name__ is set to "__main__". This allows all scripts to have a section of test code that is enabled like this...

if __name__ == "__main_":
  (test code here)

This allows any script to be tested if executed directly, and with no modifications the script can be imported into some other script and the test code section will not automatically run. 

In NumWorks Python, a generic Shell can be opened and scripts can be imported as desired, in which case all scripts have their __name__ variable set to the script's name.

But the more common way to run a single script is to select it for direct execution by selecting "Execute script" for that script. 

Executing a script in this second way is more than just an import of the script, it is also an implied execution of the script. I was surprised to discover that a script selected and executed in this way does not have its __name__ variable changed to "__main__", allowing the standard pattern to run its main code and not the main code in any imported scripts.  

When creating a set of functions to be tested standalone, to later be imported into calling scripts, the workaround with NumWorks is currently to wrap the testing code in a function, such as one named demo().  This requires the user to type "demo()" to run a script after selecting "Execute script", and is an extra step that seems unnatural, especially to a calculator user seeking simple, fast operation of a selected script.

To allow the script to go ahead and run right away, the workaround is to add a single line at the end of the script that calls demo() to run the test code. However, the user is then forced to either add or remove this line depending on whether the script is to be tested standalone or to be imported into another script.

This catch-22 situation is exactly why standard Python changes __name__ to "__main__" for a script that is being executed. All scripts can be directly tested with its own test code, yet it can be imported by other scripts for calling its functions cleanly.

My suggestion is for one small change to the way Python works in NumWorks. If a script is selected using the "Execute script" option, and only if that operation is selected, then that script's __name__ variable is set to "__main__" instead of remaining with the name of its script. All scripts, including the main one, are then imported and run exactly as they currently are. The only change is to the __name__ variable, and only in the main script. No other environment code would require changes.

The objection to this change might be that the main script is simply being imported into a shell environment, but I contend that it is also "executed", per the user's instructions when they select the "Execute script" option.  The behavior I'm suggesting would be much closer to the standard way Python scripts work when executed.

If users are using the __name__ variable in NumWorks Python in such a way that this change would break their code, then that would be important. But because the __name__ variable is not being changed, it's not very useful and I believe any such code will be non-existent. Of course I'm open to learning of any such situations!

**Comment by RedGl0w** [comment]
Indeed, epsilon doesn't execute the script, but just includes it (which can be seen when typing `__name__` in console. This behaviour should probably be changed...
